pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -261,39 +261,30 @@ func (r *SubnetSetReconciler) CollectGarbage(ctx context.Context) {
 	defer func() {
 		log.Info("SubnetSet garbage collection completed", "duration(ms)", time.Since(startTime).Milliseconds())
 	}()
-	subnetSetList := &v1alpha1.SubnetSetList{}
-	err := r.Client.List(ctx, subnetSetList)
+
+	crdSubnetSetIDs, err := r.listSubnetSetIDsFromCRs(ctx, true)
 	if err != nil {
-		log.Error(err, "Failed to list SubnetSet CR")
-		return
-	}
-	var nsxSubnetList []*model.VpcSubnet
-	for _, subnetSet := range subnetSetList.Items {
-		nsxSubnetList = append(nsxSubnetList, r.SubnetService.ListSubnetCreatedBySubnetSet(string(subnetSet.UID))...)
-	}
-	if len(nsxSubnetList) == 0 {
+		log.Error(err, "Failed to list Subnet CRs")
 		return
 	}
+	crdSubnetSetIDsSet := sets.NewString(crdSubnetSetIDs...)
 
-	subnetSetIDs := sets.New[string]()
-	for _, subnetSet := range subnetSetList.Items {
-		if err := r.deleteSubnetForSubnetSet(subnetSet, true); err != nil {
-			metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerDeleteFailTotal, MetricResTypeSubnetSet)
-		} else {
-			metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerDeleteSuccessTotal, MetricResTypeSubnetSet)
-		}
-		subnetSetIDs.Insert(string(subnetSet.UID))
-	}
-	for _, subnet := range nsxSubnetList {
-		if !r.SubnetService.IsOrphanSubnet(*subnet, subnetSetIDs) {
+	subnetSetUIDs := r.SubnetService.SubnetStore.ListIndexFuncValues(servicecommon.TagScopeSubnetSetCRUID)
```

Comment:
subnetSetUIDs -> nsxSubnetSetUIDs is better, because this is gotten from NSX resource store.

Reply:
there is a service function:
ListSubnetCreatedBySubnetSet to do same thing, we had better to call this service function rather than writing the same logic here.

Reply:
done.

Reply:
> subnetSetUIDs -> nsxSubnetSetUIDs is better, because this is gotten from NSX resource store.

After reviewing the source code, I think that seems to be the case that the `ListIndexFuncValues` returns the `SubnetSetCRUIDs`, instead of SubnetSet ID in NSX. 
For example, if there is `SubnetSet` cr-id1 in K8s, has two Subnet ns-cr-id1-xxx0, ns-cr-id-xxx1 in NSX, the `ListIndexFuncValues` will return `cr-id1`.

https://github.com/kubernetes/client-go/blob/c38ce11939a7cecae6e64542554e4de50d901919/tools/cache/thread_safe_store.go#L114
```
type Index map[string]sets.String
	func (i *storeIndex) getIndexValues(indexName string) []string {
		index := i.indices[indexName]
		names := make([]string, 0, len(index))
		for key := range index {
			names = append(names, key)
		}
		return names
	}
```